add `disableFilter` prop

kamholz opened the pull request:
This pull request adds a new `disableFilter` prop on the Svelecte component. When `true`, it disables filtering of options with the sifter in `filterList` and simply returns all of the options. This is useful in the case where the options are fetched remotely and it makes sense to filter them on the server side (for example because the dataset is large or the matching algorithm is complex). It's otherwise not really possible to achieve this functionality, as far as I can tell.

The patch is very simple but it seems to work well in my testing so far. The only strange behavior I found is that when you unfocus the input and refocus it, the dropdown reappears even with the now-empty input box, which is a bit odd for when the suggestions are correlated with the typed input. Possible fixes to deal with this would be:

* Add another new prop to control whether the input is cleared in `onBlur`, so there is an option not to do so.
* Expose an API to clear the option list, in case the desired behavior is the current default of clearing the input. For this to be useful, you might also have to dispatch an event for when the input is cleared.

There are a couple other loose ends I ran into:

* When the component has just been mounted and the input is focused, the dropdown appears with the "No options" message. Not very friendly for someone who hasn't had a chance to type anything yet!
* `highlightSearch` is very nice for the default case but there may be times you want to override it, for example when using a custom formatter that outputs text that happens to match the search string. (Simple example: imagine a custom formatter matching the value "Nathan" and outputting "Name: Nathan, Country: Germany". `highlightSearch` would erroneously highlight the "Na" in "Name".) To improve this, I suggest adding a prop `disableHighlight` and also passing `$inputValue` to custom formatters in case they want to handle the highlighting.

If you think any of these ideas are good, I can probably implement some of them before you merge anything. But I wanted to discuss it first because it goes well beyond the relatively simple change in this pull request.

By the way, feel free to rename any of these props in case you think there's a better name!

kamholz committed:
add disableFilter prop

mskocik commented:
This makes sense, it was possible in pre 1.0 version, but it was called `sortRemoteResults`. In fact you can still find this prop in default settings. Please rename the prop to `disableSifter`, because list filter itself is not really disabled and add it also to `README.md` API section and also remove `sortRemoteResults` from default settings. Thank you.

---

Regarding the other stuff - I enabled discussions, where we can discuss further features/ideas. But in short 
- handling input blur is something I have almost done, but not pushed yet. 
- 'No options' for fetch setting is bug, `i18n.fetchBefore` message should be used
- And the controlling highlighting may be desirable. We can discuss if fuirther in discussion or issue

kamholz committed:
rename disableFilter to disableSifter and remove sortRemoteResults

kamholz committed:
properly show fetchBefore message

kamholz committed:
update README.md

kamholz committed:
style fix

kamholz commented:
OK, should be good now, I think. I've renamed to `disableSifter` and updated the README. I also fixed the `fetchBefore` bug (please double-check). I might do another pull request later for the highlightSearch stuff and you can tell me what you think.

mskocik committed:
Update README.md description
